### DISCUSSION TITLE
ORB support for citar new API?

### DISCUSSION THREAD

**Discussion by LorenRiccie** [org-roam/org-roam-bibtex#253] [Q&A]
See [citar issue #675](https://github.com/emacs-citar/citar/issues/675).
what could we patch to support it?

**Comment by myshevchuk** [comment]
Hi @LorenRiccie,

we do not need to patch it. There have been changes on the Citar part, but ORB is fully compatible with them. You just need to configure Citar as described here: https://github.com/emacs-citar/citar/issues/646#issuecomment-1207205635

Note that you also need to install and load Citar helper package for Org-roam: https://github.com/emacs-citar/citar-org-roam

Your config may look as follows (I tested it and it works):

```elisp
(require 'citar-org-roam)
(citar-register-notes-source
 'orb-citar-source (list :name "Org-Roam Notes"
        :category 'org-roam-node
        :items #'citar-org-roam--get-candidates
        :hasitems #'citar-org-roam-has-notes
        :open #'citar-org-roam-open-note
        :create #'orb-citar-edit-note
        :annotate #'citar-org-roam--annotate))

(setq citar-notes-source 'orb-citar-source)
```

I will later update the ORB Readme file to reflect these innovations.

**Reply by LorenRiccie** [reply]
Thank You @myshevchuk and @bdarcus ! 
I tried this configuration
but always stuck with an  error after calling org-cite-insert:
`Debugger entered--Lisp error: (void-variable hasnotes)
  (gethash citekey hasnotes)`

**Reply by bdarcus** [reply]
That seems likely to be some config or install issue with citar and maybe org?

I don't see that error, for example, and have not seen it from anyone else.

Maybe make sure you have the latest version of citar installed and configured correctly, and if you still see it, report on the citar issue tracker?

**Reply by LorenRiccie** [reply]
Somehow using 
` (use-package citar-org-roam
   :load-path "~/.emacs.d/packages/citar-org-roam/citar-org-roam.el"
   :after citar org-roam
   :no-require
   :config (citar-org-roam-mode))`
fixed this specific error, instead of just loading the file and requiring it.

I am facing other errors now using citar with citar-org-roam that I will examine before opening a new issue. Thank You @bdarcus

**Reply by bdarcus** [reply]
I added a config section, with possible example (though in this case it's not using orb; you'd need to adjust it of course):

https://github.com/emacs-citar/citar-org-roam#configuration
